controllers/component_dependency_update_controller_test.go

```diff
@@ -402,14 +403,232 @@ var _ = Describe("Component nudge controller", func() {
+				component2.Annotations[CustomRenovateConfigMapAnnotation] = componentRenovateConfigName.Name
+				err = k8sClient.Update(context.TODO(), &component2)
+				Expect(err).ToNot(HaveOccurred())
+				waitComponentAnnotationExists(component2Name, CustomRenovateConfigMapAnnotation)
+			}
+
+			createBuildPipelineRun("test-pipeline-1", UserNamespace, BaseComponent, imageBuiltFrom)
+			Eventually(func() bool {
+				pr := getPipelineRun("test-pipeline-1", UserNamespace)
+				return controllerutil.ContainsFinalizer(pr, NudgeFinalizer)
+			}, timeout, interval).WithTimeout(ensureTimeout).Should(BeTrue())
+			pr := getPipelineRun("test-pipeline-1", UserNamespace)
+			pr.Status.SetCondition(&apis.Condition{
+				Type:               apis.ConditionSucceeded,
+				Status:             "True",
+				LastTransitionTime: apis.VolatileTime{Inner: metav1.Time{Time: time.Now()}},
+			})
+			pr.Status.Results = []tektonapi.PipelineRunResult{
+				{Name: ImageDigestParamName, Value: tektonapi.ResultValue{Type: tektonapi.ParamTypeString, StringVal: "sha256:12345"}},
+				{Name: ImageUrl, Value: tektonapi.ResultValue{Type: tektonapi.ParamTypeString, StringVal: "quay.io.foo/bar:latest"}},
+			}
+			pr.Status.CompletionTime = &metav1.Time{Time: time.Now()}
+			Expect(k8sClient.Status().Update(ctx, pr)).Should(BeNil())
+			Eventually(func() bool {
+				// check that no nudgeerror event was reported
+				failureCount := getRenovateFailedEventCount()
+				// check that renovate config was created
+				renovateConfigsCreated := len(getRenovateConfigMapList())
+				// check that renovate pipeline run was created
+				renovatePipelinesCreated := len(getRenovatePipelineRunList())
+				return renovatePipelinesCreated == 1 && renovateConfigsCreated == 2 && failureCount == 0
+			}, timeout, interval).WithTimeout(ensureTimeout).Should(BeTrue())
+		}
+
+		It("Test build performs nudge on success, only namespace wide renovate config provided", func() {
+			customConfigType := "json"
+			customConfigMapName := types.NamespacedName{Namespace: UserNamespace, Name: NamespaceWideRenovateConfigMapName}
+			automergeValue := true
+			automergeTypeValue := "pr"
+			platformAutomergeValue := false
+			ignoreTestsValue := true
+			commitMessagePrefixValue := "msg_prefix"
+			commitMessageSuffixValue := "msg_suffix"
+			imageBuiltFrom := "repo/sha"
+			commitBodyValue := fmt.Sprintf("Image created from '%s'\n\nSigned-off-by:", imageBuiltFrom)
+			fileMatchValue := []string{"cfile1", "cfile2"}
+			customConfigMapData := map[string]string{
+				RenovateConfigMapAutomergeKey:           "true",
+				RenovateConfigMapCommitMessagePrefixKey: commitMessagePrefixValue,
+				RenovateConfigMapCommitMessageSuffixKey: commitMessageSuffixValue,
+				RenovateConfigMapFileMatchKey:           "cfile1, cfile2",
+				RenovateConfigMapAutomergeTypeKey:       "pr",
+				RenovateConfigMapPlatformAutomergeKey:   "false",
+				RenovateConfigMapIgnoreTestsKey:         "true",
+			}
```

Comment:
Can it be common as well?

Reply:
it could but I don't want to separate it from output checks, this way inputs & outputs are visible in the same place

Reply:
ok, makes sense